test/cli/test-other.py

```diff
@@ -825,3 +826,41 @@ def test_file_duplicate_2(tmpdir):
         '3/3 files checked 0% done'
     ]
     assert stderr == ''
+
+
+def test_premium_with_relative_path(tmpdir): #12254
+
+
+    test_file = os.path.join(tmpdir, 'test.c')
+    with open(test_file, 'wt'):
+        pass
+
+    product_name = 'Cppcheck Premium ' + str(time.time())
+
+    test_cfg = lookup_cppcheck_exe()
+    if(test_cfg.endswith('.exe')):
+        test_cfg = test_cfg[:-4]
+    test_cfg += '.cfg'
+    with open(test_cfg, 'wt') as f:
+        f.write("""
+                {
+                    "addons": [],
+                    "productName": "NAME",
```

Comment:
Use ` {}` placeholder and call `format()` on the string instead. You can also get rid of the `product_name` variable if you do not use it in any other way and just fold it in here.

Reply:
it's also used for an assert on line 863.

Reply:
> it's also used for an assert on line 863.

I knew I overlooked it somehow.

Still you should use `format()` instead of `replace()`.